framework/CodeInterfaces/SERPENT/SerpentInterface.py

```diff
@@ -0,0 +1,191 @@
+          if index == None:
+            index = idx
+        continue
+      todo += ' '+flag
+      for ext in exts:
+        idx,fname = getFileWithExtension(inputFiles,ext.strip('.'))
+        todo+=' '+fname.getFilename()
+        if index == None:
+          index = idx
+    #outputs
+    #FIXME I think if you give multiple output flags this could result in overwriting
+    self.caseName = inputFiles[index].getBase()
+    outfile = 'out~'+self.caseName
+    if 'output' in clargs.keys():
+      todo+=' '+clargs['output']+' '+outfile
+    #text flags
+    todo+=' '+clargs['text']
+    #postpend
+    todo+=' '+clargs['post']
+    returnCommand = [('parallel',todo)],outfile
+    print('Execution Command: '+str(returnCommand[0]))
+    return returnCommand
+
+  def createNewInput(self,currentInputFiles,origInputFiles,samplerType,**Kwargs):
+    """
+      This method is used to generate an input based on the information passed in.
+      @ In, currentInputFiles, list,  list of current input files (input files from last this method call)
+      @ In, oriInputFiles, list, list of the original input files
+      @ In, samplerType, string, Sampler type (e.g. MonteCarlo, Adaptive, etc. see manual Samplers section)
+      @ In, Kwargs, dictionary, kwarded dictionary of parameters. In this dictionary there is another dictionary called "SampledVars"
+            where RAVEN stores the variables that got sampled (e.g. Kwargs['SampledVars'] => {'var1':10,'var2':40})
+      @ Out, newInputFiles, list, list of newer input files, list of the new input files (modified and not)
+    """
+    indexes=[]
+    infiles=[]
+    origfiles=[]
+    #FIXME possible danger here from reading binary files
+    for index,inputFile in enumerate(currentInputFiles):
+      if inputFile.getExt() in self.getInputExtension():
+        indexes.append(index)
+        infiles.append(inputFile)
+    for index,inputFile in enumerate(origInputFiles):
+      if inputFile.getExt() in self.getInputExtension():
+        origfiles.append(inputFile)
+    parser = GenericParser.GenericParser(infiles)
+    parser.modifyInternalDictionary(**Kwargs)
+    parser.writeNewInput(currentInputFiles,origfiles)
+    return currentInputFiles
+
+
+  def finalizeCodeOutput(self, command, output, workDir):
+    # filename would be 'input.serpent'
+    filename = command.strip().split(' ')[-1]
+    # filename_without_extension would be 'input'
+    filenameWithoutExtension = output.split('~')[1]
+    # resfile would be 'input.serpent_res.m'
+    # this is the file produced by RAVEN
+    resfile = os.path.join(workDir, filename+"_res.m")
+    inputFile = os.path.join(workDir, filename)
+    inbumatfile = os.path.join(workDir, filename+".bumat0")
+    outbumatfile = os.path.join(workDir, filename+".bumat1")
+    # get the list of isotopes to track
+    scriptLoc = os.path.dirname(os.path.realpath(sys.argv[0]))
+    isofile = os.path.join(scriptLoc, 'CodeInterfaces/SERPENT/aux-input-files/isoFile')
```

Comment:
can you explain to me this thing here? are you forcing the Serpent User to have the isoFile in ```CodeInterfaces/SERPENT/aux-input-files/```?? And if so, why?

Reply:
My intention was that the user can customize what isotopes to track by using the `isoFile`, and from the `isoFile` the user can generate the `feature_space` and the `target_space`, using the `generateCustomVariable.py`. Thinking back this sounds really buggy, if there is a work around this please let me know. Thanks!